src/docfx/config/ConfigLoader.cs

```diff
@@ -64,7 +72,7 @@ where glob(file)
             var preloadConfig = JsonUtility.ToObject<PreloadConfig>(errors, preloadConfigObject);
 
             // Download dependencies
-            var credentialProvider = preloadConfig.GetCredentialProvider();
+            var credentialProvider = preloadConfig.GetCredentialProvider(getCredential);
```

Comment:
>getCredential [](start = 73, length = 13)

`credentialRefresher` might be a better name

Reply:
To make the contract more general, it can be used to get credential in the future not only refresh credential

---
In reply to: [551897867](https://github.com/dotnet/docfx/pull/6962#discussion_r551897867) [](ancestors = 551897867)